apps/vault-apy/src/APYSimulator.ts

```diff
@@ -95,75 +137,125 @@ export default class APYSimulator {
     startingDate: Date = new Date()
   ) {
     const startingTimestamp = startingDate.getTime() / 1000;
+    const vaultData = this.#config.vaults.find(v => v.address.toLowerCase() === vaultAddress.toLowerCase());
+
+    if (!vaultData) {
+      throw new Error(`Vault address ${vaultAddress} not found in config`);
+    }
+
     for (let i = 0; i <= numOfDays; i++) {
       log(`processing day ${i}`);
-      const forkBlock = await this.#getBlockAtTimestamp(
+      let forkBlock = await this.#getBlockAtTimestamp(
         startingTimestamp - i * ONE_DAY_IN_SECONDS
       );
 
+      if (vaultData.rewardPoolType === RewardPoolType.Aura || vaultData.rewardPoolType === RewardPoolType.ConvexMainnet) {
```

Comment:
Without doing a diff, it looks like these adjustments are the same as above, I would put it into a function and re-use the code. If it's different, it's still worth breaking out into a function.